xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -934,8 +931,8 @@ void CPVRClients::ProcessMenuHooks(int iClientID, PVR_MENUHOOK_CAT cat, const CF
       if (selection >= 0)
       {
         itrClients = clients.begin();
-        for (int i = 0; i < selection; i++)
-          itrClients++;
+        for (int i = 0; i < selection; ++i)
+          ++itrClients;
         iClientID = itrClients->first;
```

Comment:
If one needs to access the map at an entry determined by a run variable, map is the wrong datatype in the first place.